dvl beta integration doc (#167)
bluerobotics/ardusub-gitbook: developers/dvl-integration.md

# DVL integration (BETA)

The DVL integration with ArduSub is in a BETA stage and is currently unsupported for normal use. This document is provided to instruct beta testers and developers how to test the BETA stage integration. There is no guarantee that the DVL integration will work for any particular application.

## Supported DVLs

The [Waterlinked A50 DVL](https://waterlinked.com/product/dvl-a50/) is the only supported DVL at the moment in the beta integration.

### A50 DVL configuration

> Set it up to use the [Static IP address](https://waterlinked.github.io/docs/dvl/dvl-a50-details/#network-config) 192.168.2.94

Author: Does this conflict with WL UGPS?

The DVL should be configured with a static ip address of 192.168.2.94. Follow Water Linked's instructions here.

Reviewer: YES! that should be .95, my mistake.

# DVL integration (BETA)

The DVL integration with ArduSub is in a BETA stage and is currently unsupported for normal use. This document is provided to instruct beta testers and developers how to test the BETA stage integration. There is no guarantee that the DVL integration will work for any particular application.

## Supported DVLs

The [Waterlinked A50 DVL](https://waterlinked.com/product/dvl-a50/) is the only supported DVL at the moment in the beta integration.

### A50 DVL configuration

> Set it up to use the [Static IP address](https://waterlinked.github.io/docs/dvl/dvl-a50-details/#network-config) 192.168.2.95

Author: ```suggestion
The DVL must be configured with a static IP address of 192.168.2.95. Follow Water Linked's instructions here.
```

 - Make sure the ROV is connected to the internet
 - Navigate to http://192.168.2.2:2770/git
 - Click "origin" -> "branches" -> "dvl"
 - Click "update"
 - Wait until the update is over and the companion reboots
 - Refresh the page and "Water Linked" should show up in the navbar

### DVL configuration in Companion

The DVL driver introduces a new interface to the http://192.168.2.2:2770/waterlinked page.

<img src="/images/waterlinked-dvl.png" class="img-responsive img-center" style="max-height:600px;">

**Driver Enable:**

Enables or disables the DVL Driver

**DVL IP Adress:**

Inital IP where the driver will try to find the DVL. The driver always attempts to find Waterlinked-dvl.local (mDNS hostname) in the local network, if that fails, it falls back to the IP address in this field.

**Orientation:**

Orientation of the DVL: **Down** is the suggested use, where the DVL points to the ocean floor, **Forward** is an untested mode where the dvl is mounted forward to lock position to a vertical surface in front of it.

**Use as Rangefinder:**

Allows the DVL to be used as a rangefinder (shown in QGC).

**Restart DVL Service:**

Restarts the DVL service.

**Set New Origin:**

> Used for dead reckoning, This makes the ROV show up in QGC at the selected location. This is likely to be not as good as the DVL's own dead reckoning.

Author: ```suggestion
Used for dead reckoning; this makes the ROV show up in QGC at the selected location. The position displayed in QGC is calculated by the autopilot independently of the position showed in the Water Linked DVL web interface, which is calculated by the DVL itself.

**Status:**

Shows the current Driver status, useful for troubleshooting.

## ArduSub Firmware

The DVL data is sent by the VISUAL_ODOMETRY mavlink message. This message is supported in ArduSub 4.1-beta and onward. The ArduSub 4.1-beta firmware is recommended for testing with DVL.

> The Firmware **must** be downloaded from the ardupilot servers [here](https://firmware.ardupilot.org/Sub/beta/Pixhawk1/ardusub.apj)

Author: ```suggestion

```
